### ISSUE TITLE
Only 9700 parameters and 13700 samples result in OOM on a TITAN Xp

### ISSUE THREAD

**Issue by Imorton-zd** [danielegrattarola/keras-gat#22]
__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input_1 (InputLayer)            (None, 600)          0                                            
__________________________________________________________________________________________________
dropout_1 (Dropout)             (None, 600)          0           input_1[0][0]                    
__________________________________________________________________________________________________
input_2 (InputLayer)            (None, 13708)        0                                            
__________________________________________________________________________________________________
graph_attention_1 (GraphAttenti (None, 16)           9648        dropout_1[0][0]                  
                                                                 input_2[0][0]                    
__________________________________________________________________________________________________
dropout_6 (Dropout)             (None, 16)           0           graph_attention_1[0][0]          
__________________________________________________________________________________________________
graph_attention_2 (GraphAttenti (None, 7)            133         dropout_6[0][0]                  
                                                                 input_2[0][0]                    
==================================================================================================
Total params: 9,781
Trainable params: 9,781
Non-trainable params: 0
__________________________________________________________________________________________________
Train on 13708 samples, validate on 13708 samples

ResourceExhaustedError (see above for traceback): OOM when allocating tensor with shape[13708,13708]
	 [[Node: graph_attention_2/leaky_re_lu_3/mul = Mul[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"](graph_attention_2/leaky_re_lu_3/Const, graph_attention_2/leaky_re_lu_3/Relu)]]
	 [[Node: loss/add_8/_151 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_2492_loss/add_8", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

**Comment by danielegrattarola** [comment]
Unfortunately this implementation of GAT does not support sparse tensors, and therefore it will not work for large graphs.
